src/main/kotlin/org/rust/lang/core/resolve/NameResolution.kt

```diff
@@ -1025,13 +1025,22 @@ private fun processLexicalDeclarations(
         return processAll(boundNames, processor)
     }
 
-    fun processCondition(condition: RsCondition?, processor: RsResolveProcessor): Boolean {
-        if (condition == null || condition == cameFrom) return false
-        val pat = condition.pat
+    fun processPatList(patList: List<RsPat>, processor: RsResolveProcessor): Boolean {
+        if (Namespace.Values !in ns) return false
```

Comment:
Looks like this line should be placed under `is RsMatchArm -> {` for uniformity.

Reply:
But it will force us to add a similar line into `processCondition`

Reply:
This line is executed before each call of `processCondition`

Reply:
Dropped redundant namespace checks